src/SonarScanner.MSBuild.TFS.Classic/Program.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.Common.Interfaces;
+using SonarScanner.MSBuild.TFS.Classic.XamlBuild;
+
+namespace SonarScanner.MSBuild.TFS.Classic
+{
+    public static class Program
+    {
+        public static int Main(string[] args)
+        {
+            var logger = new ConsoleLogger(includeTimestamp: false);
+            return Execute(args, logger);
+        }
+
+        public static int Execute(string[] args, ILogger logger)
+        {
+            try
+            {
+                if (args.Length < 1)
+                {
+                    logger.LogError("No argument found. Exiting...");
+                    Environment.ExitCode = 0;
+                    return 0;
+                }
+
+                var method = (Method)Enum.Parse(typeof(Method), args[0], false);
+                var legacyTeamBuildFactory = new LegacyTeamBuildFactory(logger);
+                var teamBuildSettings = TeamBuildSettings.GetSettingsFromEnvironment(logger);
+                AnalysisConfig config = null;
+
+                switch (method)
+                {
+                    case Method.ConvertCoverage:
+                        config = AnalysisConfig.Load(args[1]);
+                        ExecuteCoverageConverter(logger, config, legacyTeamBuildFactory, teamBuildSettings, args[2]);
+                        break;
+                    case Method.SummaryReportBuilder:
+                        config = AnalysisConfig.Load(args[2]);
+                        ExecuteReportBuilder(logger, config, legacyTeamBuildFactory, teamBuildSettings, bool.Parse(args[1]), args[3]);
+                        break;
+                }
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("An exception occured while executing the process : " + ex.Message);
+                logger.LogError(ex.StackTrace);
+            }
+
+            Environment.ExitCode = 0;
+            return 0;
```

Comment:
Why does this exe never return a "failed" exit code if something goes wrong? Is that going to make it harder to diagnose cases where something has gone wrong, like code coverage suddenly disappearing?

Reply:
I was wondering if we have to do so. We are still logging warn/errors in cas of in the code, but just never return a error exit code. What would be the benefit of that ?

Reply:
Generally speaking, returning an error code is more flexible because it means the caller can decide whether to continue or not based on the situation, if there are multiple situations/callers then they can be handled differently.

Here, it depends; are there any circumstances in which it would make sense to fail the build if an exception occurred?

Reply:
To me, coverage is an "optional" asset and shall not fail the build, even more since we rely on the third party CodeCoverage.exe which may also fail and we cannot have control over it. WDYT ?

Reply:
I think I've been asking the wrong question :-) What I should have asked is - is this a change to the existing behaviour, and if so what's the reason for the change (e.g. user feedback)?

Reply:
I think that previously we were failing the whole build if anything happened during that bit, so yes, there's a change of behavior. No specific feedback from customers/users, just my own opinion as explained above.

Reply:
If you're changing the behaviour it would be worth creating a ticket for it so it's traceable - it'll appear in the release notes, and it will make life easier for support and customers